@navikt/core/react/src/util/debounce.ts

```diff
@@ -1,25 +1,28 @@
 "use client";
+
+import { Timeout } from "./hooks/useTimeout";
+
 // https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/debounce.js
 export default function debounce<T extends unknown[]>(
   func: (...args: T) => void,
   wait = 166,
   leading = false,
 ) {
-  let timeout: ReturnType<typeof setTimeout> | undefined;
+  const timeout = new Timeout();
   function debounced(this: any, ...args: T) {
     const later = () => {
-      timeout = undefined;
+      // Clear first to mark idle before invoking callback
+      timeout.clear();
       func.apply(this, args);
     };
-    if (!timeout && leading) {
+    if (!timeout.isStarted() && leading) {
       later();
     }
-    clearTimeout(timeout);
-    timeout = setTimeout(later, wait);
+    timeout.start(wait, later);
   }
 
   debounced.clear = () => {
-    clearTimeout(timeout);
+    timeout.clear();
   };
```

Comment:
I think we can just do this

```suggestion
  debounced.clear = timeout.clear;
```